core/state.py

```
@@ -0,0 +1,163 @@
+        if address not in self.accounts:
+            self.accounts[address] = {
+                'balance': 0,
+                'nonce': 0,
+                'code': None,
+                'storage': {}
+            }
+        return self.accounts[address]
+
+    def verify_transaction_logic(self, tx):
+        if not tx.verify():
+            logger.error(f"Error: Invalid signature for tx from {tx.sender[:8]}...")
```

Ruff (0.15.0)

[warning] Logging statement uses f-string

(G004)